### ISSUE TITLE
Thunderbird fails to launch in unstable branch

### ISSUE THREAD

**Issue by imadnyc** [NixOS/nixpkgs#281827]
### Describe the bug
Thunderbird doesn't launch on nixos-unstable revision 842d9d80cfd4560648c785f8a4e6f3b096790e19. However, if I go back to 23.11, this works fine. 

### Steps To Reproduce
1. Launch thunderbird via CLI

### Expected behavior

Expected thunderbird to launch

### Additional context
Things I did:

- Launch Thunderbird via CLI
	- Error Logs
	  ```
	  ExceptionHandler::GenerateDump cloned child 32636                                                                                                                                                       
	  ExceptionHandler::SendContinueSignalToChild sent continue signal to child                                                                                                                               
	  ExceptionHandler::WaitForContinueSignal waiting for continue signal..

 	  ```
- Click "Tell Mozilla about this crash so they can fix it"
	- After a bit of waiting, get this error
	  ```
  	  Failed to open curl lib from binary, use libcurl.so instead
	  ```
	- Thunderbird still says that it sucessfully sent the report

Also, I deleted ~/.thunderbird, and the same crash happens. Again, 23.11 works fine. 

Running in safe mode, doing the two checkboxes (each permuatation) also crashes with the same errors. However, continue in troubleshoot mode works. Not sure how to troubleshoot further tbh. 

### Notify maintainers
@lovesegfault @pierron @vcunat

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
➜  ~ nix-shell -p nix-info --run "nix-info -m"                                                                                                                                                          
 - system: `"x86_64-linux"`                                                                                                                                                                             
 - host os: `Linux 6.1.72, NixOS, 23.11 (Tapir), 23.11.20240115.b8dd8be`                                                                                                                                
 - multi-user?: `yes`                                                                                                                                                                                   
 - sandbox: `yes`                                                                                                                                                                                       
 - version: `nix-env (Nix) 2.18.1`                                                                                                                                                                      
 - channels(root): `"nixos-22.11"`                                                                                                                                                                      
 - nixpkgs: `/nix/store/yy19v2dwb8ldphvia9smajvwv3ycx2c1-source`  
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by chewblacka** [comment]
Just as a point of reference thunderbird seems to launch fine for me, but I'm on unstable (24.05.20240117.842d9d8). So I think your issue is that you are running 23.11, but pulling in thunderbird from unstable, and it isn't launching?

**Comment by imadnyc** [comment]
Yes, this is the case. However, shouldn't this not affect it launching? Is there a library that isn't packaged with thunderbird that it's trying to access?

**Comment by imadnyc** [comment]
I've built my entire system against unstable, and it does seem to launch correctly.

**Comment by chewblacka** [comment]
> I've built my entire system against unstable, and it does seem to launch correctly.

ok, that is strange. I'm not sure what the issue could be.

**Comment by vcunat** [comment]
Sometimes you get issues when you mix significantly different versions of OS and packages.  Usually it's triggered by impurities like OpenGL/mesa or some other plugin mechanisms.  Upstream devs typically assume that all "visible" package versions are consistent...

**Comment by imadnyc** [comment]
Is there an effective way of finding impurities? Or is it just trial and error?

**Comment by vcunat** [comment]
I'm not aware of a nice way.

**Comment by MrKoberman** [comment]
If I run Thunderbird in safe mode it opens up, but still cant find the fix for this.

**Comment by MrKoberman** [comment]
This two tickets might be connected - https://github.com/NixOS/nixpkgs/issues/231347

**Comment by lunik1** [comment]
I have the same issue, but I do mix the unstable and stable channels. I'm also having an issue with `floorp` segfaulting on launch, which could be connected.

**Comment by lunik1** [comment]
Getting same behaviour from firefox too. Bisected it to the #281828 `staging-next` merge.

**Comment by jmarmstrong1207** [comment]
Unstable works fine for me. Is this still a problem for everyone, or can this issue be closed?

**Comment by lunik1** [comment]
I have stopped mixing the stable and unstable channels

**Comment by MrKoberman** [comment]
I still have this issue with unstable.

**Comment by jmarmstrong1207** [comment]
> I still have this issue with unstable. And when using stable, screen sharing does not work.

Are the logs the exact same as the original poster's?

**Comment by imadnyc** [comment]
@jmarmstrong1207 This is still an issue, though like @lunik1 I've just stopped mixing the stable and unstable channels. Thunderbird has so many dependencies I can't figure out what's impure. The curl bug is also peculiar, I can't tell what the issue is. From what I can tell, it _should_ be pulling curl from the derivation.
